planning/obstacle_avoidance_planner/src/utils/debug_utils.cpp

```diff
@@ -850,6 +858,22 @@ visualization_msgs::msg::MarkerArray getDebugVisualizationWallMarker(
       getVirtualWallTextMarkerArray(virtual_wall_pose, "virtual_wall_text", 1.0, 1.0, 1.0),
       &vis_marker_array);
   }
+
+  if (debug_data.stop_pose_by_drivable_area_with_margin) {
+    const auto virtual_wall_pose_with_margin =
+      getVirtualWallPose(debug_data.stop_pose_by_drivable_area_with_margin.get(), vehicle_param);
+
+    appendMarkerArray(
+      getVirtualWallMarkerArray(
+        virtual_wall_pose_with_margin, "virtual_wall_with_margin", 0.0, 1.0, 0),
+      &vis_marker_array);
+
+    appendMarkerArray(
+      getVirtualWallTextMarkerArray(
+        virtual_wall_pose_with_margin, "virtual_wall_text_with_margin", 0.0, 0.0, 1.0),
+      &vis_marker_array);
+  }
+
```

Comment:
we dont need extra markers. Instead lets rearrange this part with `stop_pose_by_drivable_area_with_margin` so the virtual wall would be only where we actually stop:

https://github.com/autowarefoundation/autoware.universe/blob/231f70c1d4920fe3c5bd822bf969a7669d284aad/planning/obstacle_avoidance_planner/src/utils/debug_utils.cpp#L844-L846